### PR TITLE
Fix incompatibility error with `--pdb` despite using `-n0`

### LINKED ISSUE #937: Plugin raises error with `--pdb` option despite using `-n0`
```toml
# pyproject.toml
[tool.pytest.ini_options]
addopts = "--dist=load"
```

```commandline
$ pytest  --pdb -n0
ERROR: --pdb is incompatible with distributing tests; try using -n0 or -nauto.
```

As you can see,  `-n0` is already passed. However, pytest still suggests using `-n0` to address the issue.

#### Solution

`-n0` should imply `--dist=no`.

### PR DESCRIPTION
**Pull request by atugushev**
Fixes #937.

### CODE REVIEW

**Comment by nicoddemus** [comment]
Rebased.

GitHub does not always run the GH actions for first time contributors, IIRC there were people opening random PRs which would use the actions for mining... 👀

File: testing/test_plugin.py [hunk 1]
```diff
@@ -48,6 +48,23 @@ def test_dist_options(pytester: pytest.Pytester) -> None:
...
+    check_options(config)
+    assert config.option.dist == "no"
+    assert config.option.tx == []
+
+    config = pytester.parseconfigure(
+        "--numprocesses", "0", "--dist", "each", "--tx", "2*popen"
+    )
+    check_options(config)
+    assert config.option.dist == "no"
+    assert config.option.tx == []
```
Comment (nicoddemus) [inline testing/test_plugin.py:66]: Don't we want to also test passing `-n0` and `--pdb` explicitly?

Comment (atugushev) [reply testing/test_plugin.py:66]: Surely we do, thanks! Addressed in 0de6180.

**Review by nicoddemus — APPROVED** [review]
Thanks!

**Pull request merged**